src/linker/Linker/LinkerAttributesInformation.cs

```diff
@@ -12,42 +13,63 @@ namespace Mono.Linker
 {
 	readonly struct LinkerAttributesInformation
 	{
-		readonly Dictionary<Type, List<Attribute>> _linkerAttributes;
+		readonly Dictionary<Type, List<object>> _linkerAttributes;
```

Comment:
I thought about this - and I'm slightly in favor of forcing the Attribute here. If we need a special version for the linker (like in your case) just make the internal linker class derive from Attribute - it doesn't really hurt anything. I think it's good to have the intent of storing structural information here - with object it feels like it's fine to store just simple values - which I think we should avoid (we would not get rid of allocation as they would be boxed anyway, and having the extra metadata is a good thing in my mind)

Reply:
Forcing it to derive from `Attribute` feels weird, but I see your point - and agree that it doesn't really hurt. I changed it back to `Attribute`.